### PR TITLE
test: improving unit tests assumptions

### PR DESCRIPTION
**Pull request by wei3erHase**
Closes BAL-62

### CODE REVIEW

File: test/unit/BPool.t.sol [hunk 1]
```diff
@@ -1434,9 +1431,8 @@ contract BPool_Unit_JoinPool is BasePoolTest {
   }
 
   function _assumeHappyPath(JoinPool_FuzzScenario memory _fuzz) internal pure {
-    vm.assume(_fuzz.initPoolSupply >= INIT_POOL_SUPPLY);
-    vm.assume(_fuzz.poolAmountOut >= _fuzz.initPoolSupply);
-    vm.assume(_fuzz.poolAmountOut < type(uint256).max / BONE);
+    _fuzz.initPoolSupply = bound(_fuzz.initPoolSupply, INIT_POOL_SUPPLY, type(uint256).max / BONE);
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:1434]: `/ BONE` is safe here, as `poolAmountOut` needs to be bigger than `initPoolSupply`, and lower than `type(uint256).max / BONE` (which means, `initPoolSupply` needs to be lesser than  `type(uint256).max / BONE`

File: test/unit/BPool.t.sol [hunk 2]
```diff
@@ -1797,6 +1799,8 @@ contract BPool_Unit_SwapExactAmountIn is BasePoolTest {
     SwapExactAmountIn_FuzzScenario memory _fuzz
   ) public happyPath(_fuzz) {
     vm.assume(_tokenIn != VM_ADDRESS);
+    vm.assume(_tokenIn != tokenIn);
+    vm.assume(_tokenIn != tokenOut);
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:1803]: is this `!= address(0)`?
can we implement `notFoundryAddress` here?

Comment (0xAustrian) [reply test/unit/BPool.t.sol:1803]: We can implement it now, it will be implemented when merging #28 anyway

Comment (wei3erHase) [reply test/unit/BPool.t.sol:1803]: #28 is for after `dev > main` PR

**Review by wei3erHase — COMMENTED** [review]
👏 

(@0xAustrian blz approve!)

**Comment by wei3erHase** [comment]
having local issues on `BPool_Unit_ExitswapExternAmountOut:test_Revert_MathApprox`

File: test/unit/BFactory.t.sol [hunk 3]
```diff
@@ -69,7 +70,8 @@ contract BFactory_Unit_NewBPool is Base {
    * @notice Test that msg.sender is set as the controller
    */
   function test_Set_Controller(address _randomCaller) public {
-    vm.assume(_randomCaller != VM_ADDRESS);
```
Comment (drgorillamd) [inline test/unit/BFactory.t.sol:72]: were you really hitting the cheatcode contract @wei3erHase ? It should be excluded from the fuzz dictionnary iirc

Comment (wei3erHase) [reply test/unit/BFactory.t.sol:72]: we were in some occasions, yes, is there a way of enforcing the exclusion?

**Comment by wei3erHase** [comment]
adding BAL-62 to remove all `address[] previousTokens` from fuzzed arrays, we should reduce as much as possible the fuzzed arrays (they're very resource expensive), plz address in different PR, can be left for stage 2 (after merging #30)

**Review by 0xAustrian — APPROVED** [review]
🚀

**Pull request merged**